libr/core/cconfig.c

```diff
@@ -2294,16 +2291,21 @@ static bool cb_io_cache_write(void *user, void *data) {
 	RCore *core = (RCore *)user;
 	RConfigNode *node = (RConfigNode *)data;
 	if (node->i_value) {
-		core->io->cached |= R_PERM_W;
+		core->io->cache.mode |= R_PERM_W;
 	} else {
-		core->io->cached &= ~R_PERM_W;
+		core->io->cache.mode &= ~R_PERM_W;
 	}
 	return true;
 }
 
 static bool cb_io_cache(void *user, void *data) {
-	(void)cb_io_cache_read (user, data);
-	(void)cb_io_cache_write (user, data);
+	RCore *core = (RCore *)user;
+	RConfigNode *node = (RConfigNode *)data;
+	if (node->i_value) {
+		core->io->cache.mode |= R_PERM_X;
```

Comment:
now that is merged, next steps would be:

* use the new rvec.h instead of rlists
* bring back the unit tests handling more use case
* add wc--* to remove all layers
* add the ability to name each layer (associate with stack, binfile, etc)

also i would like to discuss about the following topics:

* decide if we want to have 0 io cache layers (wc-- will remove a layer, if there was only 1 should be generate a new empty layer? or just say there are no layers?)
* a way to add a name to each layer (api + command)
* clarify the meaning of io.cache.mode